src/bindgen/ir/enumeration.rs

```diff
@@ -897,6 +897,42 @@ impl Source for Enum {
                 out.close_brace(false);
             }
 
+            if config.language == Language::Cxx
+                && config
+                    .enumeration
+                    .derive_tagged_enum_copy_constructor(&self.annotations)
+            {
+                out.new_line();
+                out.new_line();
+                write!(
+                    out,
+                    "{}(const {}& {})",
+                    self.export_name, self.export_name, other
+                );
+                out.new_line();
+                write!(out, " : tag({}.tag)", other);
+                out.open_brace();
+                write!(out, "switch (tag)");
+                out.open_brace();
+                for variant in &self.variants {
+                    if let Some((ref variant_name, ref item)) = variant.body {
+                        write!(
+                            out,
+                            "case {}::{}: new (&{}) {}(other.{}); break;",
+                            self.tag.as_ref().unwrap(),
+                            variant.export_name,
+                            variant_name,
+                            item.export_name(),
+                            variant_name,
+                        );
+                        out.new_line();
+                    }
+                }
+                write!(out, "default: break;");
```

Comment:
You should probably check if you have any non-trivial variants before generating the switch, because IIRC MSVC emits warnings & such for switch statements with only one labeled 'default' case.

Reply:
We only get here with non-trivial variants, otherwise we generate a plain `enum` instead of a struct / union.